src/migrations/20240520000000-merge_duplicate_args.js

```diff
@@ -0,0 +1,232 @@
+          PARTITION BY arid,gid
+          ORDER BY "updatedAt" DESC, arg.id
+        ) updated_rank
+      FROM "ActivityReportGoals" arg
+      JOIN link_counts
+        ON "activityReportId" = arid
+        AND "goalId" = gid
+      WHERE link_cnt > 1
+      )
+      SELECT
+        id donor_arg,
+        argid target_arg
+      FROM "ActivityReportGoals" arg
+      JOIN latest_updated
+        ON "activityReportId" = arid
+        AND "goalId" = gid
+        AND updated_rank = 1
+      ;
+
+      -- Relink any ActivityReportGoalFieldResponses connected to the
+      -- duplicate (and therefore donor) ARG
+      -- Because there could theoretically be multiple prompts on
+      -- multiple duplicates, we need to rank the ARGFRs referring to
+      -- a particular prompt-goal pair and select just one of each.
+      -- There's one target_arg per goal so we use that as a proxy.
+      -- Just for simplicity, ARGFRs that are already on the target
+      -- ARG are left alone and the corresponding responses on donors
+      -- will be deleted.
+      --
+      -- At time of writing this is all theoretical as there aren't
+      -- any reponses at all for FEI goals with duplicate ARGs, but this
+      -- could change by the time it runs
+      DROP TABLE IF EXISTS relinked_argfrs;
+      CREATE TEMP TABLE relinked_argfrs
+      AS
+      WITH updater AS (
+        WITH argfr_on_donor_args AS (
+          SELECT
+            donor_arg,
+            target_arg,
+            argfr."activityReportGoalId" argid,
+            argfr."goalTemplateFieldPromptId" promptid,
+            argfr.id argfrid,
+            ROW_NUMBER() OVER (
+              PARTITION BY target_arg, argfr."goalTemplateFieldPromptId"
+              ORDER BY argfr."activityReportGoalId" = target_arg, argfr."updatedAt" DESC, argfr.id
+            ) choice_rank
+          FROM arg_merges am
+          JOIN "ActivityReportGoalFieldResponses" argfr
+            ON am.donor_arg = argfr."activityReportGoalId"
+        ), unmatched AS (
+          SELECT
+            donor_arg,
+            argid,
+            argfrid
+          FROM argfr_on_donor_args aoda
+          WHERE choice_rank = 1
+            AND argid != target_arg
+        )
+        UPDATE "ActivityReportGoalFieldResponses" AS argfr
+        SET "activityReportGoalId" = target_arg
+        FROM arg_merges am
+        JOIN unmatched u
+          ON u.donor_arg = am.donor_arg
+        WHERE argfr.id = u.argfrid
+        RETURNING
+          id argfrid,
+          am.donor_arg original_arg
+      ) SELECT * FROM updater
+      ;
+
+      -- Delete duplicate objective ARGFRs
+      DROP TABLE IF EXISTS deleted_argfrs;
+      CREATE TEMP TABLE deleted_argfrs
+      AS
+      WITH updater AS (
+        DELETE FROM "ActivityReportGoalFieldResponses"
+        USING arg_merges
+        WHERE "activityReportGoalId" = donor_arg
+        RETURNING
+          id argfrid,
+          donor_arg
+      ) SELECT * FROM updater
+      ;
```

Comment:
Can you double check this delete. I created a duplicate response case and after running the delete I had no responses for that arg. I would think we would want to keep one of them.

Inserted duplicate records like this: 
  insert into "ActivityReportGoalFieldResponses" (id, "activityReportGoalId", "goalTemplateFieldPromptId", response, "createdAt", "updatedAt")
	  values 
	  (1584, 12355, 1, '{Workforce}','2024-05-17 12:35:45.871-04','2024-05-17 12:35:45.871-04' )
	  
	  	  insert into "ActivityReportGoalFieldResponses" (id, "activityReportGoalId", "goalTemplateFieldPromptId", response, "createdAt", "updatedAt")
	  values 
	  (1585, 12354, 1, '{Workforce}','2024-05-17 12:35:45.871-04','2024-05-17 12:35:45.871-04' )